### DISCUSSION TITLE
New inflation settings

### DISCUSSION THREAD

**Discussion by RicaProEXP** [OpenTTD/OpenTTD#12970] [Features]
There is already an inflation setting that makes prices increase over time. Make this rate customisable and allow negative numbers so that players can experiment deflation, inflation, hyperinflation.
Replace the x1.02 with x0.99

**Comment by flix1** [comment]
I'm also interested in this, for educational purposes. I can show students in Venezuela and Argentina how to deal with very high inflation... but also what long term deflation would look like. OpenTTD is the best econ teaching tool!

**Comment by wensimehrp** [comment]
You can actually change the inflation rate manually in JGRPP (a popular patchpack) but even in JGRPP it doesn't allow you to change it to a value smaller than 1

**Reply by flix1** [reply]
Thanks. 
Is that 1% annually?

**Reply by wensimehrp** [reply]
> Thanks. 
> Is that 1% annually?

1 here means 100%... Basically you cannot simulate deflation in JGRPP, only inflation and hyperinflation

**Reply by flix1** [reply]
How difficult would it be to do this? Is it just a matter of changing some parameters?

I'm thinking of hiring someone to do this specific mod... but as you can imagine my budget is limited. So if it's a weeks long project I'll just give up now and stop wasting everyone's time. But if it can be done in a few hours... I'm up for it.

**Comment by LC-Zorg** [comment]
> I'm also interested in this, for educational purposes. I can show students in Venezuela and Argentina how to deal with very high inflation... but also what long term deflation would look like. OpenTTD is the best econ teaching tool!

Recognizing this game as a good tool for learning economics seems quite controversial to me. :) In my opinion, this game is too simplified for such simulations, but maybe it wouldn't be bad to show some basics...

Honestly, I'm very curious how you would present ways to deal with this very high inflation in OTTD. :)

A few years ago I presented the idea of ​​inflation, the level of which would change during the game.
This may be obvious, but in general, there are two types of inflation in the game:
- revenue inflation
- cost inflation

Cost inflation is higher than revenue inflation by default. This is intended to gradually increase the difficulty level of the game. Inflation itself also causes a gradual devaluation of the accumulated money, which also affects the level of difficulty, although only to a small extent, because in this game the player always has excess money anyway. :)
This difference in inflation levels and the increase in difficulty cause the game to become too difficult for some players, especially when using add-ons where the maintenance costs of vehicles are high by default.
This caused a desire to introduce changes some time ago.
There was a change where inflation - if enabled - always starts at 1920. This was to help add-on creators adjust vehicle prices and costs to inflation, which would be specified and always the same. I don't know if any of the creators did it. I doubt it. Meanwhile, the real effect of this change is that a player starting the game later has a significantly increased difficulty level from the very beginning, and if they didn't use the few, if any add-ons adjusted to this change, the change, in my opinion, made the situation worse instead of improving it.
There were several proposed solutions. Among others, the ability to specify the level of both inflations. And this is exactly what was introduced in the JGR version. I don't know, but maybe you could convince him to implement factors lower than 1.00.
Personally, I'm not convinced that such settings make much sense for new games, but they could help fix those where inflation increased costs too much, so such a change would make sense in general in my opinion. :)
Returning to the idea of ​​a variable inflation level, this was also supposed to be a solution that could help reduce costs and adjust them to the player's skill level. There would also be a continuous increase in difficulty, but never above a certain limit.
The principle of this function would be quite simple:
- Generally, for simplicity, revenue inflation would be constant or possibly variable, but dependent only on the level of investment (the effect of a lot of money on the market and increased demand), with periods of increased inflation during randomly generated recessions, which could be different. This type of inflation can also be 0, and only costs can change.
- Cost inflation would be the same as revenue inflation in base, but would change depending on what costs relative to income all companies in the game are incurring. When costs are low, cost inflation would be higher than revenue inflation, which would cause the difficulty to gradually increase. When costs are very high (above a certain level), their inflation could be negative, which would make even games previously ruined by too much inflation playable again. In reality, this would follow from the principle that things are worth what the customer is willing to pay for them.

Here is the topic with the original description and the mentioned inflation problem after the change: #11738
Here is the topic with the description of the inflation problem before introduce the change: #8429
And here is description of the change itself: #7589

**Comment by flix1** [comment]
Just like **Minecraft** is good to teach some basic economic concepts, **OpenTTD** is great for teaching some others. Obviously I'm talking micro. No computer game afaik has yet done a realistic macro econ model. 

OpenTTD is great at teaching profit & loss, capital accumulation, opportunity cost, competition, prices, entrepreneurship, intro to accounting...

How inflation affects all of these at the micro level could be demonstrated very well using OpenTTD... at least at a basic level. Of course many things are unrealistic! Infinite money being one of them... that's why a deflation setting would be great for experimenting.

I was settling for the simplest possible change. If you let me... I have ideas for a totally different econ model for the game in which there are no money sinks and faucets, but all money has to be earned!

**Comment by flix1** [comment]
Thanks @LC-Zorg for the explanation on how inflation works in-game.

I guess that to keep things very simple, what I would need would be:

1. Make revenue inflation = cost inflation
2. Allow player to set the inflation % in settings
3. Allow that multiplier to be < 1. (As in annually prices are multiplied by X, where X can be between 0.50-10.00)

I'll take a look at the discussions you've linked to see if I can find someone to help me with this. Thanks!

**Comment by flix1** [comment]
Another possibility would be to use the historical inflation numbers...
This classic economic history chart (https://fraser.stlouisfed.org/files/docs/publications/1943chart_busibooms.pdf) has data from 1775 to 1944!

![image](https://github.com/user-attachments/assets/fe7abd54-751d-49ea-9efb-7e1c00882178)

Including long periods of deflation and others of inflation.

**Reply by wensimehrp** [reply]
> Another possibility would be to use the historical inflation numbers... This classic economic history chart (https://fraser.stlouisfed.org/files/docs/publications/1943chart_busibooms.pdf) has data from 1775 to 1944!
> 
> ![image](https://private-user-images.githubusercontent.com/4233784/372809308-fe7abd54-751d-49ea-9efb-7e1c00882178.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc4ODcyMjcsIm5iZiI6MTcyNzg4NjkyNywicGF0aCI6Ii80MjMzNzg0LzM3MjgwOTMwOC1mZTdhYmQ1NC03NTFkLTQ5ZWEtOWVmYi03ZTFjMDA4ODIxNzgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAwMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMDJUMTYzNTI3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YTc2NGQwMTc4YTU0ZDgxYWI5NDZjODAwMzEzMDU1MTVhNTM0NDJlNWU4M2U2MmQyMzA2YjQ0YmY4OGEwZTZmOCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ._7XBNXfRedmsVPKGcWE1IY5O0Y5CXLEjkPjJ6sbRYrM)
> 
> Including long periods of deflation and others of inflation.

In this case you could probably host a multiplayer game and use a script to send inflation settings to the server...
